File: GasProject Server/Sources/World/Objects/Taser.cpp
#include "Taser.hpp"

#include <IGame.h>
#include <World/World.hpp>
#include <World/Tile.hpp>
#include <World/Objects/Projectile.hpp>

Taser::Taser() {
    sprite = "taser";
    name = "Taser";
}

void Taser::InteractWith(Object *obj) {
	if (!obj) 
		return;

	if (!obj->GetTile() || !GetTile())
		return;

	uf::vec2i direction = (obj->GetTile()->GetPos() - GetTile()->GetPos()).xy();
	GGame->GetWorld()->CreateObject<Projectile>(GetTile(), direction);
}

File: Lua/Object/Taser.lua
local Projectile = require "Projectile"
local Object = require "Object"
derive(Object, ...)

function New(this)
	Object.New(this);
	this.sprite = "taser";
end
function InteractWith(this, o)
	if not o or not o:GetTile() or not this:GetTile() then return; end;
	Projectile(this:GetTile(), o:GetTile():GetPos() - this:GetTile():GetPos());
end
